core/common/test/LocalDateRangeTest.kt

```diff
@@ -0,0 +1,274 @@
+    fun step() {
+        assertContentEquals(
+            (1..24 step 2).map { LocalDate(2000, 1, it) },
+            (Jan_01_2000..Jan_24_2000).step(2, DateTimeUnit.DAY)
+        )
+        assertContentEquals(
+            (24 downTo 1  step 2).map { LocalDate(2000, 1, it) },
+            (Jan_24_2000 downTo Jan_01_2000).step(2, DateTimeUnit.DAY)
+        )
+    }
+
+    @Test
+    fun string() {
+        assertEquals(
+            "2000-01-01..2000-01-05",
+            (Jan_01_2000..Jan_05_2000).toString()
+        )
+        assertEquals(
+            "2000-01-05 downTo 2000-01-01 step -1D",
+            (Jan_05_2000 downTo Jan_01_2000).toString()
+        )
+        assertEquals(
+            "2000-01-01..2000-01-05 step 1D",
+            LocalDateProgression.fromClosedRange(Jan_01_2000, Jan_05_2000, 1, DateTimeUnit.DAY).toString()
+        )
+        assertEquals(
+            "2000-01-05 downTo 2000-01-01 step -1D",
+            LocalDateProgression.fromClosedRange(Jan_05_2000, Jan_01_2000, -1, DateTimeUnit.DAY).toString()
+        )
+    }
+
+    @Test
+    fun random() {
+        assertEquals(
+            Jan_01_2000,
+            (Jan_01_2000..Jan_01_2000).random()
+        )
+
+        assertEquals(
+            Jan_01_2000,
+            (Jan_01_2000 downTo Jan_01_2000).random()
+        )
+
+        assertFails {
+            (Jan_02_2000..Jan_01_2000).random()
+        }
+
+        assertNull((Jan_02_2000..Jan_01_2000).randomOrNull())
+
+        val seed = 123456
+        val expectedRand = Random(seed)
+        val actualRand = Random(seed)
+
+        repeat(20) {
+            assertEquals(
+                expectedRand.nextLong(0L..23L).let { Jan_01_2000.plus(it, DateTimeUnit.DAY) },
+                (Jan_01_2000..Jan_24_2000).random(actualRand)
+            )
+        }
+
+        repeat(20) {
+            assertEquals(
+                expectedRand.nextLong(0L..23L).let { Jan_24_2000.minus(it, DateTimeUnit.DAY) },
+                (Jan_24_2000 downTo Jan_01_2000).random(actualRand)
+            )
+        }
+
+        listOf(1L, 2L, 5L, 30L).forEach { step ->
+            repeat(20) {
+                val range = (0L..23L step step)
+                assertEquals(
+                    expectedRand.nextLong(0L..range.last / step).let { Jan_01_2000.plus(it * step, DateTimeUnit.DAY) },
+                    (Jan_01_2000..Jan_24_2000).step(step, DateTimeUnit.DAY).random(actualRand)
+                )
+            }
+
+            repeat(20) {
+                val range = (0L..23L step step)
+                assertEquals(
+                    expectedRand.nextLong(0..range.last / step).let { Jan_24_2000.minus(it * step, DateTimeUnit.DAY) },
+                    (Jan_24_2000 downTo Jan_01_2000).step(step, DateTimeUnit.DAY).random(actualRand)
+                )
+            }
+        }
+        repeat(20) {
+            assertTrue {
+                (Jan_01_2000..Jan_24_2000).step(5, DateTimeUnit.DAY).let { it.contains(it.random()) }
```

Comment:
This test is not deterministic, which I'm not thrilled about. The other way to test it would be with a spy, but adding mockk just for this one test case seemed like overkill/overstepping.

Reply:
In my opinion, this test is just fine the way it is, but if you prefer to test this deterministically, you can make a custom `Random` implementation and pass it to the function.

Reply:
In my opinion, this test is fine the way it is (in fact, we have other randomly generated things in our tests), but if you prefer to test this deterministically, you can make a custom `Random` implementation and pass it to the function.